src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -99,9 +99,7 @@ OverlayImpl::Timer::on_timer(error_code ec)
     overlay_.m_peerFinder->once_per_second();
     overlay_.sendEndpoints();
     overlay_.autoConnect();
-
-    if ((overlay_.timer_count_ % Tuning::checkIdlePeers) == 0)
```

Comment:
Why is this deleted? This was added so that we don't check for idle peers too frequently.

Reply:
`timer_count_` is [initialized](https://github.com/ripple/rippled/pull/3753/files#diff-3f3119d499efbf4f0ec803cf22831e82d48df5a396e321bb343961c920454b8eL141) to 0 and never updated. It is always 0, and this condition is always true.

Reply:
oh, right! it's a bug then! it should be incremented.

Reply:
i see what happened. it used to be this:

    if ((++overlay_.timer_count_ % Tuning::checkSeconds) == 0)
        overlay_.check();

    if ((overlay_.timer_count_ % Tuning::checkIdlePeers) == 0)
        overlay_.deleteIdlePeers();

but then two top lines were deleted in one of the unrelated commits.

Reply:
This doesn't seem like the right PR to fix that bug, but I wouldn't remove the code either. Let's just keep the code for now and open an issue so we don't lose track of this. @gregtatcam can you open an issue?

Reply:
I opened the issue #3754.